File: Leetcode 3401-3500/3432/3432.py
# Leetcode 3432: Count Partitions with Even Sum Difference
# https://leetcode.com/problems/count-partitions-with-even-sum-difference/
# Solved on 5th of November, 2025
class Solution:
    def countPartitions(self, nums: list[int]) -> int:
        """
        Counts the number of ways to partition the array into two non-empty subarrays
        such that the absolute difference between their sums is even.
        :param nums: The input list of integers.
        :return: The number of valid partitions.
        """
        arraySum = sum(nums)
        if arraySum % 2 == 0:
            return len(nums) - 1

        return 0

File: Leetcode 3401-3500/3432/3432.java
// Leetcode 3432: Count Partitions with Even Sum Difference
// https://leetcode.com/problems/count-partitions-with-even-sum-difference/
// Solved on 5th of December, 2025
class Solution {
    /**
     * Given an integer array nums, return the number of partitions with an even sum difference.
     *
     * @param nums The input integer array.
     * @return The number of partitions with an even sum difference.
     */
    public int countPartitions(int[] nums) {
        int totalSum = 0;
        for (int num : nums) {
            totalSum += num;
        }

        if (totalSum % 2 == 0) {
            return nums.length - 1;
        }
        return 0;
    }
}